DocGen4/IncludeStr.lean

```diff
@@ -7,20 +7,46 @@ import Lean
 
 namespace DocGen4
 
-open Lean System IO Lean.Elab.Term
+open Lean System IO Lean.Elab.Term FS
+
+deriving instance DecidableEq for FileType
+
+/--
+  Traverse all subdirectories fo `f` to find if one satisfies `p`.
+-/
+partial def traverseDir (f : FilePath) (p : FilePath → IO Bool) : IO (Option FilePath) := do
```

Comment:
Could this be replaced with a call to https://leanprover-community.github.io/mathlib4_docs/Init/System/IO.html#System.FilePath.walkDir ? Seems like it to me?

Reply:
Seems no. The `walkDir` function behaves quite differently: when a path `p` statisfy the `enter` it will push all its sub dir, and transversely find paths in those pushed; while for a unstatisfed `p'` it directly skip all its sub dir. 

While for this hack we need to traverse all the sub sir until we find a first match.

Although they looks similar, the `walkDir` always returns a `#[]` to me. :P

Reply:
:( Alrighty then.